### PR TITLE
fix: Preview server polish - duplicate filenames, DRY, tests (#119-121)

### LINKED ISSUE #119: Preview Server: Handle duplicate filenames in subdirectories
## Context
From PR #117 code review - potential issue with duplicate filenames in subdirectories.

## Problem
If multiple directories are scanned or recursive scanning is enabled, files with the same filename in different subdirectories will cause conflicts. The `get_by_filename()` method returns the first match, which could lead to serving the wrong file.

Example:
- `output/video.mp4`
- `output/subdir/video.mp4`

Both would have `filename='video.mp4'`, and only the first one found would be served.

## Proposed Solutions
1. Use relative paths from the base directory instead of just filename
2. Add a check/warning for duplicate filenames during scan
3. Store full path and use a unique identifier for file lookup

## Priority
Low - edge case that's unlikely in typical usage

## Related
- PR #117 review comment

### PR DESCRIPTION
**Pull request by jcaldwell1066**
## Summary

Addresses Copilot code review feedback from PR #117 by implementing three polish items for the preview server.

Fixes #119, #120, #121

### Handle Duplicate Filenames (#119)
- Add `relative_path` field to `MediaFile` for disambiguating files in subdirectories
- Add `unique_id` property that uses `relative_path` if available, otherwise `filename`
- Add `get_by_id()` method to Gallery with O(1) lookup via `_file_index` dictionary
- Update `to_dict()` to include `relative_path` and `unique_id` in JSON output

### Extract format_bytes Helper (#120)
- Move `format_bytes()` to module-level function (was duplicated)
- Export from `__init__.py` for reuse by other modules
- `MediaFile.size_human` now uses shared helper

### Comprehensive Test Coverage (#121)
- Add 22 new server tests in `test_preview_server.py`
- Add 10 new gallery tests for unique_id/relative_path functionality
- Security tests for path traversal protection
- Security tests for XSS prevention (html.escape)
- **Total: 59 passing tests**

## Security Checklist
- [x] All HTML output uses html.escape() for user data (verified by tests)
- [x] File paths validated against path traversal (verified by tests)
- [x] Large files streamed, not loaded into memory (unchanged)

## Testing
- [x] Unit tests added/updated
- [x] Integration tests for new features
- [x] Edge cases covered (empty, error, boundary)
- [x] All tests pass (59/59)

```
Ran 59 tests in 7.594s
OK
```

## Type
- [x] Bug fix (addresses code review feedback)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: atari_style/preview/gallery.py [hunk 1]
```diff
@@ -13,6 +13,23 @@
...
+        size_bytes: Size in bytes
+
+    Returns:
+        Human-readable string (e.g., "1.5 MB")
+    """
+    size = float(size_bytes)
+    for unit in ['B', 'KB', 'MB', 'GB']:
+        if size < 1024:
+            return f"{size:.1f} {unit}"
+        size /= 1024
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply atari_style/preview/gallery.py]: Fixed. Added `BYTES_PER_KB = 1024` constant and updated `format_bytes()` to use it.

File: atari_style/preview/gallery.py [hunk 2]
```diff
@@ -66,11 +86,22 @@ def resolution(self) -> str:
...
         return "-"
 
+    @property
+    def unique_id(self) -> str:
+        """Return unique identifier for this file.
+
+        Uses relative_path if available, otherwise filename.
+        This handles duplicate filenames in subdirectories.
+        """
+        return self.relative_path if self.relative_path else self.filename
```
[the remark this answers is not in the document]
Comment (jcaldwell1066) [reply atari_style/preview/gallery.py]: Fixed in fabafd9. Updated server.py to use `unique_id` and `get_by_id()` throughout:
- URLs now use `f.unique_id` instead of `f.filename`
- Lookups now use `get_by_id()` instead of `get_by_filename()`

Files with duplicate names in different subdirectories will now be correctly accessible.
